### PR TITLE
Abc plus logger

### PR DESCRIPTION
**Pull request by MickyHoltzman**

### CODE REVIEW

File: Experiment_Launcher_code/ExperimentManager.py [hunk 1]
```diff
@@ -211,7 +208,8 @@ def StateActivity(self, state, mouse1simulated, mouse2simulated):
 
         return ExperimentCompleted
 
-    def start_streaming_exp(self, num_trial, decision_time, return_time, opponent_type, opponent1_strategy, opponent2_strategy):
+    def start_streaming_exp(self, experiment_name, num_trial, decision_time, return_time, opponent_type, opponent1_strategy, opponent2_strategy):
```
Comment (Anna4142) [inline Experiment_Launcher_code/ExperimentManager.py:211]: ok

File: Experiment_Launcher_code/ExperimentManager.py [hunk 2]
```diff
@@ -27,10 +27,7 @@ def __init__(self):
         # initialize software components
         self.videoAnalyser = Video_Analyzer()
         self.stateManager = StateManager()
-        #self.data_path = 'C:/Users/EngelHardBlab.MEDICINE/Desktop/experimentfolder/PILOT_RESULTS/abcdefghi.csv'  ###change to whatever.csv
-        self.data_path = './../../ExperimentLogFiles/abcdefghi.csv'
-        # initialize trial logger
-        self.trial_logger = TrialLogger(self.data_path)
+        self.trial_logger = TrialLogger()
```
Comment (Anna4142) [inline Experiment_Launcher_code/ExperimentManager.py:30]: ok

File: Experiment_Launcher_code/ExperimentLauncher.py [hunk 3]
```diff
@@ -25,7 +25,8 @@ def main():
         # Initialize and start the experiment
         expManager = ExperimentManager()
         print("Experiment manager now running")
```
Comment (Anna4142) [inline Experiment_Launcher_code/ExperimentLauncher.py:27]: ok

File: Data_analysis/logger.py [hunk 4]
```diff
@@ -1,14 +1,19 @@
 import csv
+from datetime import datetime
 
 class TrialLogger:
-    def __init__(self, file_path):
-        self.file_path = file_path
-        self.csv_file = open(self.file_path, 'w', newline='')
+    def __init__(self):
```
Comment (Anna4142) [inline Data_analysis/logger.py:5]: ok

File: modelling_opponent/PrisonerABC.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+from abc import ABC, abstractmethod
```
Comment (Anna4142) [inline modelling_opponent/PrisonerABC.py:1]: what is the purpose of this?

Comment (MickyHoltzman) [reply modelling_opponent/PrisonerABC.py:1]: As we discussed on Thursday, this is designed to organize the interface to all opponent types. I changed the method names and arguments per our discussion.

**Review by Anna4142 — APPROVED** [review]

**Pull request merged**